src/logsumexp.jl

```diff
@@ -114,21 +114,23 @@ _logsumexp_onepass_op((xmax, r)::Tuple{<:Number,<:Number}, x::Number) =
 _logsumexp_onepass_op(x::Number, xmax::Number, r::Number) =
     _logsumexp_onepass_op(promote(x, xmax)..., r)
 function _logsumexp_onepass_op(x::T, xmax::T, r::Number) where {T<:Number}
+    # The following invariants are maintained through the reduction:
+    # `xmax` is the maximum of the elements encountered so far,
+    # ``r = ∑ᵢ exp(xᵢ - xmax)`` over the same elements.
     _xmax, _r = if isnan(x) || isnan(xmax)
         # ensure that `NaN` is propagated correctly for complex numbers
         z = oftype(x, NaN)
         z, r + exp(z)
     else
         real_x = real(x)
         real_xmax = real(xmax)
-        if real_x > real_xmax
+        if isinf(real_x) && isinf(real_xmax) && (real_x * real_xmax) > 0
```

Comment:
I think this change should be benchmarked. IIRC the performance of `logsumexp` was very sensitive to even minor changes of the implementation. To avoid the multiplication one might want to use
```suggestion
        if isinf(real_x) && isinf(real_xmax) && sign(real_x) == sign(real_xmax)
```